### PR TITLE
Use functional style.

### PR DESCRIPTION
**Pull request by johnjohndoe**
- Just a little rewrite.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -54,13 +54,11 @@ function pad(num, totalDigits) {
 function getTimeTableHtml(openingRanges) {
     var html = '<table class="times">';
     if (openingRanges !== undefined) {
-        for (var index = 0, openingRangesLength = openingRanges.length; index < openingRangesLength; ++index) {
-            var openingRange = openingRanges[index];
-
+        openingRanges.forEach(function(openingRange) {
```
Comment (ubergesundheit) [inline js/main.js]: Looks good, for maximum functional-style, you could rewrite this using `reduce` like this (not tested):
```javascript
html += openingRanges.reduce(function (tableRows, openingRange) {
  var tableRow = getTableRowForDay(openingRange, dayIsToday);
  return tableRows + getTableRowForDay(openingRange, dayIsToday);
}, '');
```

File: js/main.js [hunk 2]
```diff
@@ -204,13 +202,11 @@ function getOpeningTimes(openingHoursStrings) {
...
 function getOpeningRangeForDate(openingRanges, date) {
     if (openingRanges !== undefined) {
-        for (var index = 0, openingRangesLength = openingRanges.length; index < openingRangesLength; ++index) {
-            var openingRange = openingRanges[index];
-
-            var dayIsToday = openingRangeMatchesDay(openingRange, date);
-            if (dayIsToday) {
-                return openingRange;
-            }
+        var found = openingRanges.filter(function(openingRange) {
```
[the remark this answers is not in the document]
Comment (ubergesundheit) [reply js/main.js]: This shouldn't be `pending`.. :)

[the remark this answers is not in the document]
Comment (johnjohndoe) [reply js/main.js]: @ubergesundheit Please explain.

File: js/main.js (hunk 1 above)
Comment (johnjohndoe) [reply js/main.js]: Resolved in eefc320992052378b6fbbbc0bc11259bad517d75.

**Comment by johnjohndoe** [comment]
@ubergesundheit I force pushed a new version of the branch. Please take a look.

File: js/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ubergesundheit) [reply js/main.js]: Ah, I didn't read until line 214. 

The return value of `some` is boolean, but you're using the value of found in 214.

This is perfectly fine like it is!

If I understand correctly, another way could be:

```javascript
return openingRandes.find(function (openingRange) {
  return openingRangeMatchesDay(openingRange, date)
});
```

[the remark this answers is not in the document]
Comment (johnjohndoe) [reply js/main.js]: Cool. I changed it in 3f675a2ac6ef77762e9e0ab4a2bd43b541e9a444. Is this what you thought off?

[the remark this answers is not in the document]
Comment (ubergesundheit) [reply js/main.js]: Looks good, especially the fat arrow function. 

I haven’t had time to look at the code style here, but checking for === undefined an then returning early would reduce intendation. But this is really not necessary, it should work fine as it is!

**Pull request merged**